### PR TITLE
Token-based authorization.

### PR DESCRIPTION
**Pull request by mgcam**
Token-based authorization, initially for the pipeline
endpoints only, the pipeline the token belongs to is not
validated in the context of the request.

### CODE REVIEW

**Review by nerdstrike — CHANGES_REQUESTED** [review]
I'm generally fine with this. What is it you intend to happen to permissions on get_pipeline?

Might be interesting to see if we could make a suitable `@decorator` for admin routes

File: server/npg/porchdb/auth.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        self.logger = logging.getLogger(__name__)
+
+    async def token2permission(self, token: str):
+
+        message = None
+
+        if len(token) != AUTH_TOKEN_LENGTH:
+            message = f"The auth token should be {AUTH_TOKEN_LENGTH} chars long"
+        else:
+            prog = re.compile(r'\w{32}', flags=re.ASCII)
```
[the remark this answers is not in the document]
Comment (mgcam) [reply server/npg/porchdb/auth.py]: Improved

**Comment by mgcam** [comment]
> What is it you intend to happen to permissions on get_pipeline?
Any valid token should work.

> Might be interesting to see if we could make a suitable `@decorator` for admin routes
We do not have many at the moment, will leave this for future

**Review by nerdstrike — APPROVED** [review]
I like this a lot more now!

**Comment by nerdstrike** [comment]
Redudant with #33

**Pull request closed without merging**